src/chunkserver/uri_paser.h

```diff
@@ -32,29 +32,51 @@ namespace chunkserver {
 
 class UriParser {
  public:
-    static std::string ParseUri(const std::string& uri,
-                             std::string * param) {
+    // Other error codes can be added later.
+    enum ParseUrlErrorCode{
+        SUCESS = 0,
+        EPROTOCALEMPTY = 1,
+        EPROTOCALINVAILD = 2,
+    };
+
+    static ParseUrlErrorCode ParseUri(const std::string& uri,
```

Comment:
I think you have changed this function purpose. The previous version parses a Uri, and corresponding protocol and parameters are returned by return value and argument separately, also if the Uri is invalid, the return value is empty. But the after version returns parameters and whether the protocol is valid, the protocol type is lossed.